### PR TITLE
Modify rule S2055(Java): Update rule according to the LayC

### PR DESCRIPTION
**Pull request by alban-auzeill**
image
image
image

### CODE REVIEW

File: rules/S2055/java/rule.adoc [hunk 1]
```diff
@@ -1,61 +1,163 @@
 == Why is this an issue?
```
Comment (leveretka) [inline rules/S2055/java/rule.adoc:1]: I think in this section you still haven't explained, why this is an issue. I think it's worth mentioning, what happens if you don't follow this rule. Like that there will be an exception at runtime.

File: rules/S2055/java/rule.adoc [hunk 2]
```diff
@@ -1,61 +1,163 @@
...
 
+public class OtherSerializableClass extends SerializableClass {
+  // is also serializable because it is a subtype of Serializable
+}
+----
 
-=== Noncompliant code example
+Serialization stores the fields of serializable classes and ignores the fields of non-serializable classes.
+Deserialization instantiates serializable objects and fields without using any declared constructors. And for the
+non-serializable objects, it calls the implicit or explicit no-argument constructor.
```
Comment (leveretka) [inline rules/S2055/java/rule.adoc:25]: I find this section more confusing than the previous description if you don't know what it is about. How about:

"Serializable classes can have non-serializable ancestors, and to initialize fields inherited from such ancestor the no-argument constructor will be invoked. If there's no implicit or explicit no-argument constructor, the error occurs at runtime".

**Review by leveretka — CHANGES_REQUESTED** [review]

File: rules/S2055/java/rule.adoc (hunk 2 above)
Comment (alban-auzeill) [reply rules/S2055/java/rule.adoc]: @leveretka I have reworked the rule and updated the above screen capture.

**Review by leveretka — APPROVED** [review]
LGTM!

**Pull request merged**